client/bootstrap.js

```diff
@@ -0,0 +1,17 @@
+window.module = {}
+window.__NEXT_DATA__ = JSON.parse(
```

Comment:
This can be done in `client/index.js` right 🤔

Reply:
No. There’s a function appended to the beginning of each page.js and they load async so they could be called at any moment.

Reply:
```
window.__NEXT_DATA__ = JSON.parse(
```

I meant reading the `__NEXT_DATA__`, the rest still has to be defined beforehand like you said.

Reply:
@timneutkens Becuase `__NEXT_DATA__` is used in the lines following: https://github.com/zeit/next.js/pull/4943/files#diff-358a769020e11c021478d8f344f90a7eR11